sdk/common/src/main/scala/org/typelevel/otel4s/sdk/Attributes.scala

```diff
@@ -16,49 +16,40 @@
 
 package org.typelevel.otel4s.sdk
 
-import cats.Applicative
 import cats.Hash
-import cats.Monad
 import cats.Monoid
 import cats.Show
 import cats.implicits._
 import org.typelevel.otel4s.Attribute
 import org.typelevel.otel4s.Attribute.KeySelect
 import org.typelevel.otel4s.AttributeKey
 
-/** An immutable collection of [[Attribute]]s.
+/** An immutable collection of [[Attribute]]s. It contains only unique keys.
+  *
+  * @see
+  *   [[https://opentelemetry.io/docs/specs/otel/common/#attribute-collections]]
   */
-final class Attributes private (
-    private val m: Map[AttributeKey[_], Attribute[_]]
-) {
-  def get[T: KeySelect](name: String): Option[Attribute[T]] = {
-    val key = KeySelect[T].make(name)
-    m.get(key).map(_.asInstanceOf[Attribute[T]])
-  }
-  def get[T](key: AttributeKey[T]): Option[Attribute[T]] =
-    m.get(key).map(_.asInstanceOf[Attribute[T]])
-
-  def isEmpty: Boolean = m.isEmpty
-  def size: Int = m.size
-  def contains(key: AttributeKey[_]): Boolean = m.contains(key)
-  def foldLeft[F[_]: Monad, B](z: B)(f: (B, Attribute[_]) => F[B]): F[B] =
-    m.foldLeft(Monad[F].pure(z)) { (acc, v) =>
-      acc.flatMap { b =>
-        f(b, v._2)
-      }
-    }
-  def forall[F[_]: Monad](p: Attribute[_] => F[Boolean]): F[Boolean] =
-    foldLeft(true)((b, a) => {
-      if (b) p(a).map(b && _)
-      else Monad[F].pure(false)
-    })
-  def foreach[F[_]: Applicative](f: Attribute[_] => F[Unit]): F[Unit] =
-    m.foldLeft(Applicative[F].unit) { (acc, v) =>
-      acc *> f(v._2)
-    }
+sealed trait Attributes extends Iterable[Attribute[_]] {
```

Comment:
what's the reason for changing from a `final class` to a `sealed trait`?

Reply:
We may need another implementation that takes [attribute limits](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/README.md#attribute-limits) into account. 

With `Attributes` being a sealed trait, it should be easier to deal with multiple implementations. And we primarily use `sealed trait` across the otel4s code base.